assets/app/view/game/part/track_node_path.rb

```diff
@@ -138,7 +138,7 @@ def calculate_regions(edge0, edge1, arc, exit0)
           # assume can never have edge1=edge, edge0=non-edge
           if !edge0
             CENTER
-          elsif edge0.to_f != edge0.to_i.to_f
+          elsif edge0.is_a?(Float)
```

Comment:
https://github.com/tobymao/18xx/pull/4653#commitcomment-48517992

Reply:
> How does rubocop feel about changing this to elsif edge0 % 1 != 0?

It’s no float comparison, so it should be fine with it. I’ll test.

Reply:
No complaints as expected.

Reply:
Diffs of /tiles/all:
```
diff tiles_dev.html tiles_prod.html
2c2
< <html class=" nyrttqcl idc0_329">
---
> <html class=" ridzssbyml idc0_329">
9,16c9,16
<     <link rel="stylesheet" href="tiles_dev_files/normalize.css">
<     <link rel="stylesheet" href="tiles_dev_files/css2.css">
<     <link id="favicon_svg" rel="icon" type="image/svg+xml" href="http://localhost:9292/images/icon.svg">
<     <link id="favicon_32" rel="icon" type="image/png" sizes="32x32" href="http://localhost:9292/images/favicon-32x32.png">
<     <link id="favicon_16" rel="icon" type="image/png" sizes="16x16" href="http://localhost:9292/images/favicon-16x16.png">
<     <link id="favicon_apple" rel="apple-touch-icon" href="http://localhost:9292/apple-touch-icon.png">
<     <link rel="mask-icon" href="http://localhost:9292/images/mask.svg" color="#f0e68c">
<     <link rel="manifest" href="http://localhost:9292/site.webmanifest">
---
>     <link rel="stylesheet" href="tiles_prod_files/normalize.css">
>     <link rel="stylesheet" href="tiles_prod_files/css2.css">
>     <link id="favicon_svg" rel="icon" type="image/svg+xml" href="https://18xx.games/images/icon.svg">
>     <link id="favicon_32" rel="icon" type="image/png" sizes="32x32" href="https://18xx.games/images/favicon-32x32.png">
>     <link id="favicon_16" rel="icon" type="image/png" sizes="16x16" href="https://18xx.games/images/favicon-16x16.png">
>     <link id="favicon_apple" rel="apple-touch-icon" href="https://18xx.games/apple-touch-icon.png">
>     <link rel="mask-icon" href="https://18xx.games/images/mask.svg" color="#f0e68c">
>     <link rel="manifest" href="https://18xx.games/site.webmanifest">
21c21
<     <link rel="stylesheet" href="tiles_dev_files/main.css">
---
>     <link rel="stylesheet" href="tiles_prod_files/main.css">
39,40c39,40
<             <h1 id="logo" style="margin: 0px; font-size: 1rem; white-space: nowrap;"><a href="http://localhost:9292/" title="18xx.Games" style="color: currentcolor; font-weight: bold; text-decoration: none;"><span style="display: inline-block; height: 3rem; width: 2.5rem; line-height: 3rem; background: rgba(0, 0, 0, 0) url(&quot;/images/logo_polygon_yellow.svg&quot;) no-repeat scroll left center / 2.5rem; color: rgb(0, 0, 0); text-align: center;">18xx</span><span> . Games</span></a></h1>
<             <div id="nav" style="margin: auto 0px;"><a href="http://localhost:9292/about" style="margin: 0px 0.5rem;">About</a><a href="http://localhost:9292/profile" style="margin: 0px 0.5rem;">Profile (yzemaze)</a></div>
---
>             <h1 id="logo" style="margin: 0px; font-size: 1rem; white-space: nowrap;"><a href="https://18xx.games/" title="18xx.Games" style="color: currentcolor; font-weight: bold; text-decoration: none;"><span style="display: inline-block; height: 3rem; width: 2.5rem; line-height: 3rem; background: rgba(0, 0, 0, 0) url(&quot;/images/logo_polygon_yellow.svg&quot;) no-repeat scroll left center / 2.5rem; color: rgb(0, 0, 0); text-align: center;">18xx</span><span> . Games</span></a></h1>
>             <div id="nav" style="margin: auto 0px;"><a href="https://18xx.games/about" style="margin: 0px 0.5rem;">About</a><a href="https://18xx.games/profile" style="margin: 0px 0.5rem;">Profile (yze-test1)</a></div>
16067,16068c16067,16068
<         <script type="text/javascript" src="tiles_dev_files/deps.js"></script>
<         <script type="text/javascript" src="tiles_dev_files/main.js"></script>
---
>         <script type="text/javascript" src="tiles_prod_files/deps.js"></script>
>         <script type="text/javascript" src="tiles_prod_files/main.js"></script>
16071c16071
<     Opal.$$.App.$attach("app", Opal.hash("app_route", "/tiles/all", "production", false, "user", Opal.hash("id", 194, "name", "yzemaze", "email", "yzemaze@example.com", "settings", Opal.hash("bg", "#000000", "bg2", "#dcdcdc", "red", "#ec232a", "blue", "#35a7ff", "font", "#ffffff", "gray", "#bcbdc0", "brown", "#cb7745", "font2", "#000000", "green", "#71bf44", "white", "#eae0c8", "yellow", "#fde900", "consent", true, "r0_dash", "0", "r1_dash", "0", "r2_dash", "0", "r3_dash", "0", "r4_dash", "0", "r5_dash", "0", "r6_dash", "0", "r7_dash", "0", "r0_color", "#a40122", "r0_width", "8", "r1_color", "#008df9", "r1_width", "8", "r2_color", "#00fccf", "r2_width", "8", "r3_color", "#ff5aaf", "r3_width", "8", "r4_color", "#8400cd", "r4_width", "8", "r5_color", "#ff6e3a", "r5_width", "8", "r6_color", "#ffb2fd", "r6_width", "8", "r7_color", "#ffc33b", "r7_width", "8", "red_logo", false, "webhooks", [Opal.hash("url", "https://hooks.slack.com/foobar", "name", "slack"), Opal.hash("url", "https://hooks.discord.com/foobar", "name", "discord"), Opal.hash("url", "bar", "name", "foo")], "your_turn", "#ff8c00", "hotseat_game", "#af8cff", "notifications", false))))
---
>     Opal.$$.App.$attach("app", Opal.hash("app_route", "/tiles/all", "production", true, "user", Opal.hash("id", 1925, "name", "yze-test1", "email", "18xx-test1@yzemaze.de", "settings", Opal.hash("bg", "#000000", "bg2", "#dcdcdc", "red", "#ec232a", "blue", "#35a7ff", "font", "#ffffff", "gray", "#bcbdc0", "brown", "#cb7745", "font2", "#000000", "green", "#71bf44", "white", "#eae0c8", "yellow", "#fde900", "consent", true, "r0_dash", "0", "r1_dash", "0", "r2_dash", "0", "r3_dash", "0", "r4_dash", "0", "r5_dash", "0", "r6_dash", "0", "r7_dash", "0", "bg_color", "#333333", "r0_color", "#a40122", "r0_width", "8", "r1_color", "#008df9", "r1_width", "8", "r2_color", "#00fccf", "r2_width", "8", "r3_color", "#ff5aaf", "r3_width", "8", "r4_color", "#8400cd", "r4_width", "8", "r5_color", "#ff6e3a", "r5_width", "8", "r6_color", "#009f81", "r6_width", "8", "r7_color", "#ffc33b", "r7_width", "8", "red_logo", false, "your_turn", "#ff8c00", "email_sent", 1610205664, "font_color", "#d7d7d7", "hotseat_game", "#af8cff", "simple_logos", true, "notifications", true, "last_password_reset", 1614417121))))
``

So at least there shouldn’t be any difference in tile rendering.

Reply:
LGTM, I'll leave this conversation unresolved for now so @roseundy can chime in if he wants

Reply:
I don't think this is correct. The code is trying to determine if an edge is an integer, because an edge like 1.5 is legal and renders next to the vertex between edges 0 and 1.

Reply:
Why are we changing this again?

Reply:
> Why are we changing this again?

https://docs.rubocop.org/rubocop/1.0/cops_lint.html#lintfloatcomparison

Reply:
https://docs.rubocop.org/rubocop/cops_lint.html#lintfloatcomparison discourages doing equality comparisons with floats; we're changing to put more cops to work

the old line here was `elsif edge0.to_f != edge0.to_i.to_f`, the latest line (changed since my first comment so this isn't what you see at the top of this comment chain) is `elsif edge0 % 1 != 0`, I think it's got the correct behavior:

![Screenshot 2021-03-21 123020](https://user-images.githubusercontent.com/1045173/111916585-429c9f00-8a41-11eb-9cc8-ce02fb66f8f9.png)

Reply:
> I don't think this is correct. The code is trying to determine if an edge is an integer, because an edge like 1.5 is legal and renders next to the vertex between edges 0 and 1.

`edge0.to_f != edge0.to_i.to_f`
… is true for floats ( with % 1 != 0), but false for integers. Hence the change to `elsif edge0 % 1 != 0`